tests/core/test_funcs.py

```diff
@@ -0,0 +1,73 @@
+import uuid
```

Comment:
Please could the structure of this file match the structure of the source tree?
i.e. tests for `maths.py` should go in `test_maths.py` and tests for `coordination.py` should go in `test_coordination.py`.